workshop/06-datasource.md

```diff
@@ -0,0 +1,115 @@
+? Connection String url to override other settings (eg: https://username:password@host): https://your-username-here:your-password-here@your-host-here
+? database: album
+? username:
+? password:
+? modelIndex:
+```
+
+And finally, let's change our `album` model to use the Cloudant DB. We can do so by updating our `model-config.json`. We'll see a section near the bottom for `album` and will update the object key 'dataSource' to 'cloudantDB', which is what we called this data-source when adding it to our application above. It should look like this:
+
+```
+  "album": {
+    "dataSource": "cloudantDB",
+    "public": true
+  }
+```
+
+*Note: if you are storing your username and password in your code, you will want to be sure to NOT commit them to a public repository.*
+
+In an effort to address this concern, let's dive in a little deeper.
+
+## Hidden secrets
+
+The easiest way to contain sensitive information in your application is to keep it in a file that is in your 'ignore list', in this instance, our `.gitignore`. LoopBack has a way of handling this pretty easily -- the application, on startup, looks for a number of files that aren't there by default, but can be added to handle special circumstances, such as these. The file we will use is `datasources.local.js` and it will expose our secrets file, which we can call `env.js`.
+
+## `env.js`
+
+In `env.js`, we export our data-source configuration object with our secrets.
+
+*Note: we must remember to add this `env.js` file to our `.gitignore` so that it doesn't accidentally end up in our public repository. Please do so now.*
+
+```javascript
+'use strict';
+
+module.exports = {
+  cloudantDB: {
+    "url": "https://your-username:your-password@your-host",
+    "database": "album",
+    "username": "",
+    "password": "",
+    "name": "cloudantDB",
+    "modelIndex": "",
+    "connector": "cloudant"
+  }
+};
+```
+
+## `datasources.local.js`
+
+Our `datasources.json` file is a JSON file that has no JavaScript functionality. So we have to put that logic in our `datasources.local.js`. From this file, we can simply require in the secret `env.js` file (stored in the root of our application) and export it through Node's module system. This will make the `cloudantDB` object we exported in the `env.js` file available to the system. LoopBack looks at this particular file with the expectation that if it exists, it will merge it with the data-source JSON config file, `datasources.json`.
+
+```javascript
+'use strict';
+
+module.exports = require('../env.js');
+```
+
+## `datasources.json`
+
+This file needs to exist for the application to run, so we can either make it an empty file or simply leave the config object for our in-memory data-source even though we currently aren't using it.
+
+```json
+{
+  "db": {
+    "name": "db",
+    "connector": "memory"
+  }
+}
+```
+
+Okay, now that we have our secrets hidden, let's move on.
+
+**Next Step:** [Connect a datasource](06-datasource.md)
```

Comment:
This may have been on purpose (I assume the next page hasn't been written), but you have a circular link here.